events/recorder_histogram.go

```diff
@@ -48,18 +48,26 @@ func (r *histogramStream) IncError(val int) {
 func (r *histogramStream) Record(dur time.Duration) {
 	r.catcher.Add(r.point.Counters.Number.RecordValue(1))
 	r.catcher.Add(r.point.Timers.Duration.RecordValue(int64(dur)))
+	if !r.started.IsZero() {
+		r.catcher.Add(r.point.Timers.Total.RecordValue(int64(time.Since(r.started))))
+	}
 }
+
 func (r *histogramStream) Begin() {
-	if r.started.IsZero() {
-		r.catcher.Add(r.collector.Add(r.point))
-	} else {
-		r.catcher.Add(r.point.Timers.Total.RecordValue(int64(time.Since(r.started))))
+	if r.point.Timestamp.IsZero() {
+		r.point.Timestamp = r.started
+	}
+
+	if !r.started.IsZero() {
+		r.catcher.Add(r.collector.Add(*r.point))
+		r.point.Timestamp = time.Time{}
```

Comment:
We did that on line 57, but we definitly want to reset it after we collect so that the next time it's correct. This allows SetTime to do the right thing.